### ISSUE TITLE
Exception tracking

### ISSUE THREAD

**Issue by axsuul** [cantino/huginn#1957]
Sometimes my webhook agents error out cause of some error based on the content (i.e. `UndefinedConversionException`). I don't want to miss out on events due to errors. Is there a recommendation on how I could gracefully handle errors to my webhooks or can I propose we add support for exception tracking (Sentry, Airbrake, etc).

**Comment by dsander** [comment]
Hi @axsuul,

I thought about this a few times as well. We currently have no way to track exceptions. I can think of two possible solutions. Either add a way to get notified whenever an agent log entry is created or add one of the exception tracking service you mentioned.

**Comment by cantino** [comment]
We could make an agent that receives the errors from other agents and then you could have it send to an email agent, perhaps? ;)

**Comment by axsuul** [comment]
@cantino Yea that would be awesome!

On a sidenote, is there a way to trigger an event if a WebhookAgent does not receive an event after a certain amount of time? (monitoring cron jobs)?

**Comment by cantino** [comment]
@axsuul, I think you could probably use the GapDetectorAgent.

**Comment by joenas** [comment]
How about including something like this: https://github.com/ankane/errbase

Since `Agent#error` only takes a string as an argument I would suggest adding it [here](https://github.com/huginn/huginn/blob/master/app/jobs/agent_check_job.rb#L10) and [here](https://github.com/huginn/huginn/blob/master/app/jobs/agent_receive_job.rb#L11)!

**Comment by dsander** [comment]
I like that approach, the biggest challenge I see is that we need to add configuration blocks for all suported error notification services to make them configurable via the `.env` file.

**Comment by joenas** [comment]
Ah well that takes some work if all of them should be included. Maybe just leave it to the user? Put things like `config/initializers/rollbar.rb` in `.gitignore`?

**Comment by albertsun** [comment]
While I'm normally a big fan of exception tracking services, I think here I would be more in favor of something internal to Huginn where all errors within an Agent get caught and sent to another agent / which can then alert people. Maybe we could add an option to the Agent model to "report exceptions" where it'll email the user that created the agent?

Huginn can have users who are not Huginn developers who would want to get notified of errors. Using errbase, Airbrake, etc., really makes error reporting only useful for developers.

**Comment by joenas** [comment]
Fair enough! Maybe we could do both, for me Rollbar is great since I already got it for other apps I run, with custom notifications (like slack) already setup (not a fan of email..). 
But then again, if the "ErrorAgent" could emit Events I could just send them to a SlackAgent.
